src/simulate/main.py

```diff
@@ -42,31 +46,35 @@ def main():
 
     files = os.listdir(json_files_dir)
     logging.info('list of json files in the directory %s', files)
-    for file in files:
-        path_to_file = json_files_dir + "/" + file
-        try:
-            caliper_event = open(path_to_file, 'rb')
-        except IOError as e:
-            logging.error('cannot read the file %s due to %s', path_to_file, e)
-            continue
-        with caliper_event:
-            event = caliper_event.read()
+    for i in range(run_count):
+        for file in files:
+            path_to_file = json_files_dir + "/" + file
             try:
-                jsonEvent = json.loads(event)
-            except JSONDecodeError as e:
-                logging.error('Failed to Deserialize the caliper event %s ', e)
+                caliper_event = open(path_to_file, 'rb')
+            except IOError as e:
+                logging.error('cannot read the file %s due to %s', path_to_file, e)
                 continue
-            # make needed changes to the json events
-            event_transformer = Transformer(jsonEvent, config_yml_obj)
-            json_event_transformed = event_transformer.event_transformer()
+            with caliper_event:
+                event = caliper_event.read()
+                try:
+                    jsonEvent = json.loads(event)
+                except JSONDecodeError as e:
+                    logging.error('Failed to Deserialize the caliper event %s', e)
+                    continue
+                # make needed changes to the json events
+                event_transformer = Transformer(jsonEvent, config_yml_obj)
+                json_event_transformed = event_transformer.event_transformer()
 
-            if json_event_transformed is None:
-                logging.error('Problem in transforming a event Json')
-                continue
-            # sending to endpoint
-            handler = HttpHandler(config_yml_obj)
-            handler.make_api_call(json_event_transformed)
-    logging.info('End Of App')
+                if json_event_transformed is None:
+                    logging.error('Problem in transforming a event Json')
+                    continue
+                # sending to endpoint
+                handler = HttpHandler(config_yml_obj)
+                handler.make_api_call(json_event_transformed)
+                sleep(random.uniform(0.5, 1))
```

Comment:
Here's an idea:  I know the sleep is a small amount of time, but rather than actually sleeping, how about just adding a random amount to a timestamp and sending it without delay?

(Events in the endpoint may appear to be from the future this way, but that's probably not a bad thing.)

Reply:
I have address you idea here https://github.com/tl-its-umich-edu/caliper-event-analysis/pull/15/files#r149469668